### ISSUE TITLE
Standard Deviation, Percentile Ranks and Percentiles metric error in Kibana 7.15.2

### ISSUE THREAD

**Issue by Akhil-OoBA** [fbaligand/kibana-enhanced-table#239]
Hi @fbaligand ,@cpiment ,
We are facing error while creating enhanced table visualisation that is having standard deviation metric in it.

Kibana Version: 7.15.2
Enhanced table Version: v1.12.2

Steps to reproduce: 
1) Create a new enhanced table visualisation using  "kibana_sample_data_ecommerce" index-pattern.
2) Select the 'Standard Deviation' metric aggregation.
3) Select the field  "day_of_week_i" and update.

Here is the detail log message:
TypeError: Cannot read property 'type' of undefined
    at new AggConfigResult (http://localhost:5601/44266/bundles/plugin/enhancedTable/1.12.2/enhancedTable.plugin.js:1239:17)
    at http://localhost:5601/44266/bundles/plugin/enhancedTable/1.12.2/enhancedTable.plugin.js:1610:35
    at Array.map (<anonymous>)
    at http://localhost:5601/44266/bundles/plugin/enhancedTable/1.12.2/enhancedTable.plugin.js:1609:32
    at Array.map (<anonymous>)
    at splitTable (http://localhost:5601/44266/bundles/plugin/enhancedTable/1.12.2/enhancedTable.plugin.js:1608:18)
    at enhancedTableResponseHandler (http://localhost:5601/44266/bundles/plugin/enhancedTable/1.12.2/enhancedTable.plugin.js:1652:13)
    at ExpressionFunction.fn (http://localhost:5601/44266/bundles/plugin/enhancedTable/1.12.2/enhancedTable.plugin.js:1940:28)

Thanks

**Comment by fbaligand** [comment]
After some tests, the problem concerns not only Kibana 7.15, but also Kibana 7.12 and above.
I will try to fix it in next release.

**Comment by Akhil-OoBA** [comment]
Thanks for the response, May I know when can expect the next release?

Thanks

**Comment by fbaligand** [comment]
Sorry, I can’t give you a date. 
but I will add a comment here when the release will be available.

**Comment by cpiment** [comment]
Hi,

I have made some tests in Kibana 7.15.2 and modifying line 98 of `enhanced-table-request-handler.js` from this:

```js
    column.aggConfig = aggs.byId(column.meta.sourceParams.id;
```

to this:

```js
    column.aggConfig = aggs.byId(column.meta.sourceParams.id.split(".")[0]);
```

Seems to to the trick. The problem is that the "Standard Deviation" aggregation is a metric that creates two columns, but since the aggregation is only one, the two columns are assigned IDs `"1.std_lower"` and `"1.std_upper"`, but the aggConfig ID is just `"1"`, so to find the aggConfig the ID of the column must be extracted from the first part of the column ID. 

I think that this error happens for all the metrics that create more than one column, such as Percentile Ranks or Percentiles, but this changes solve that too. This also works with aggregations that create only one column such as Count or Average.

I have not made a PR with this change because I do not know how to open it, should I open it on the 7.12 tag or on the master branch? What do you think @fbaligand?

**Comment by fbaligand** [comment]
Well done @cpiment !
You can open the PR on master branch. 
Then, I will backport the fix on other branches (7.12, 7.14)

**Comment by fbaligand** [comment]
Great news: Enhanced Table v1.13.0 is out with this issue fixed!
Numerous other enhancements are also available.

Check it there:
https://github.com/fbaligand/kibana-enhanced-table/releases/tag/v1.13.0
